CONTRIBUTING.md

```diff
@@ -25,6 +25,52 @@ The following information is most helpful:
+* **README.md** — User-facing documetnation for the extension
+* **CHANGELOG.md** — Version history and changes
+* **CONTRIBUTING.md** — This file
+* **LICENSE** — License information about the project
+* **NOTICE.txt** — License information about third party packages
+* **client** — Main extension folder
+    * **extension.ts** — Code insertion point.  Contains initialization code and defines commands callable by user.
+    * **marklogicClient.ts** — Interface to MarkLogic [Node Client API](https://github.com/marklogic/node-client-api) package, which is used to communicate with ML Server.
+    * **marklogic-types.d.ts** — TypeScript type declarations used by the package.
+* **server** — Language server extension folder (provides syntax highlighting; based on vscode-languageserver extension)
+    * **package.json** — As the language server is technically a separate package, this defines the dependencies for it
+    * **server.ts** — Code insertion point.  Contains initialization code for language server.
+
+### Building and Testing Project
+
+Building the project requires [Node.js](https://nodejs.org/) to be installed on your local machine
+
+First, dev dependencies must be installed.  Run `npm install` in both this folder, and in the `./server` folder.
+
+***IMPORTANT NOTE:***
+*The shell scripts used to drive node compilation assume a *nix environment.  If you are on Windows, you have a few options:*
+* Install WSL
+* Install GitBash
+* Install PowerShell v7+
+    * Windows Powershell v5.1 is not supported
+    * [Migration Instructions](https://learn.microsoft.com/en-us/powershell/scripting/whats-new/migrating-from-windows-powershell-51-to-powershell-7?view=powershell-7.2)
+    * Note: this is currently broken because of [an open bug in npm](https://github.com/npm/cli/issues/5332).  Use Bash.  :(
+        
+You can choose which environment should be used as the default shell for npm with the command `npm config set script-shell [shell]`.  For instance, to use WSL Bash, you would run `npm config set script-shell bash`.  PowerShell v7+ is `npm config set script-shell pwsh`  You can reset this with `npm config delete script-shell`.  Even if running npm from PowerShell, it will try to use cmd.exe as the default shell if you do not define it.
+
+To compile changes, run `npm run-script compile`.  Note that both the client and the language server need to finish compiling before you attempt to run the extension, or you will get an error.  Other script definitions are in `package.json.`
+
+It is recommended to use VSCode as the editor for this package, as it can self-load a debug instance.
+
+* Open the project folder in VSCode
+* Select the "Run and Debug" sidebar window (Ctrl+Shift+D)
+* Choose "Launch Extension (debug)" from the RUN AND DEBUG dropdown menu
+* Press the green play button or F5 to compile and launch the plugin in a test environment
```

Comment:
These steps are really good, but I think we need to tell the developer how to run the equivalent of a "hello world" - otherwise, I'm just looking at a new VS Code instance with a blank screen, wondering what to do. 

For example, I opened a new editor and entered `console.log("Hi")` and then knew to select "MarkLogic: Eval JS" from the command palette, but I get this error:

```
{
  "message": "eval XQuery on server: cannot process response with 500 status",
  "statusCode": 500,
  "body": {
    "errorResponse": {
      "statusCode": 500,
      "status": "Internal Server Error",
      "messageCode": "INTERNAL ERROR",
      "message": "XDMP-RWEMPTY: The value of expression '$0' is required to be non-empty in rule: set-database . See the MarkLogic server error log for further detail."
    }
  }
}
```

I checked that my default ML settings - localhost/8000/admin/admin - are valid. I tried highlighting the text and calling the command, but no luck. 

Hopefully this is just a config issue for me, but regardless - this guide should provide a "hello world" example so that the developer knows their test environment is working correctly.

Reply:
I just checked, and yes - it's a bug.  The Documents database needs to be set.  I'll add a note in the Readme, remove the misleading text in the config options and create a new issue for it.

Reply:
https://github.com/marklogic-community/mlxprs/issues/109